File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/callback/ParametersAndReturnsTestPercent.jl
println("getPercent")
if isdefined(Main, :getPercent) && !isnothing(getPercent)
	error("getPercent already defined!")
end
getPercent = callback("getPercent")
println(typeof(getPercent))
println(getPercent)
if typeof(getPercent) != Float64
	error("getPercent not Float64!")
end
callback("setPercent",getPercent)

println("getPercentVector")
if isdefined(Main, :getPercentVector) && !isnothing(getPercentVector)
	error("getPercentVector already defined!")
end
getPercentVector = callback("getPercentVector")
println(eltype(getPercentVector))
println(getPercentVector)
if eltype(getPercentVector) != Float64
	error("getPercentVector not Float64!")
end
callback("setPercentVector",getPercentVector)

println("getPercentVectorAsList")
if isdefined(Main, :getPercentVectorAsList) && !isnothing(getPercentVectorAsList)
	error("getPercentVectorAsList already defined!")
end
getPercentVectorAsList = callback("getPercentVectorAsList")
println(eltype(getPercentVectorAsList))
println(getPercentVectorAsList)
if eltype(getPercentVectorAsList) != Float64
	error("getPercentVectorAsList not Float64!")
end
callback("setPercentVectorAsList",getPercentVectorAsList)

println("getPercentMatrix")
if isdefined(Main, :getPercentMatrix) && !isnothing(getPercentMatrix)
	error("getPercentMatrix already defined!")
end
getPercentMatrix = callback("getPercentMatrix")
println(eltype(getPercentMatrix))
println(getPercentMatrix)
if eltype(getPercentMatrix) != Float64
	error("getPercentMatrix not Float64!")
end
callback("setPercentMatrix",getPercentMatrix)

println("getPercentMatrixAsList")
if isdefined(Main, :getPercentMatrixAsList) && !isnothing(getPercentMatrixAsList)
	error("getPercentMatrixAsList already defined!")
end
getPercentMatrixAsList = callback("getPercentMatrixAsList")
println(eltype(getPercentMatrixAsList))
println(getPercentMatrixAsList)
if eltype(getPercentMatrixAsList) != Float64
	error("getPercentMatrixAsList not Float64!")
end
callback("setPercentMatrixAsList",getPercentMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestPercent.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getPercent")
if(getPercent ~= nil) then
	error("getPercent already defined!")
end
getPercent = callback("getPercent")
getPercentType = type(getPercent)
print(getPercentType)
print(getPercent)
if(getPercentType ~= "number") then
	error("getPercent not Double!")
end
callback("setPercent", getPercent)

print("getPercentVector")
if(getPercentVector ~= nil) then
	error("getPercentVector already defined!")
end
getPercentVector = callback("getPercentVector")
getPercentVectorType = type(getPercentVector[1])
print(getPercentVectorType)
print(arraysToString(getPercentVector))
if(getPercentVectorType ~= "number") then
	error("getPercentVector not Double!")
end
callback("setPercentVector", getPercentVector)

print("getPercentVectorAsList")
if(getPercentVectorAsList ~= nil) then
	error("getPercentVectorAsList already defined!")
end
getPercentVectorAsList = callback("getPercentVectorAsList")
getPercentVectorAsListType = type(getPercentVectorAsList[1])
print(getPercentVectorAsListType)
print(arraysToString(getPercentVectorAsList))
if(getPercentVectorAsListType ~= "number") then
	error("getPercentVectorAsList not Double!")
end
callback("setPercentVectorAsList", getPercentVectorAsList)

print("getPercentMatrix")
if(getPercentMatrix ~= nil) then
	error("getPercentMatrix already defined!")
end
getPercentMatrix = callback("getPercentMatrix")
getPercentMatrixType = type(getPercentMatrix[1][1])
print(getPercentMatrixType)
print(arraysToString(getPercentMatrix))
if(getPercentMatrixType ~= "number") then
	error("getPercentMatrix not Double!")
end
callback("setPercentMatrix", getPercentMatrix)

print("getPercentMatrixAsList")
if(getPercentMatrixAsList ~= nil) then
	error("getPercentMatrixAsList already defined!")
end
getPercentMatrixAsList = callback("getPercentMatrixAsList")
getPercentMatrixAsListType = type(getPercentMatrixAsList[1][1])
print(getPercentMatrixAsListType)
print(arraysToString(getPercentMatrixAsList))
if(getPercentMatrixAsListType ~= "number") then
	error("getPercentMatrixAsList not Double!")
end
callback("setPercentMatrixAsList", getPercentMatrixAsList)
